src/utilities.py

```diff
@@ -256,22 +257,20 @@ def check_item(self, item):
 
 
 def string_to_float(string: str):
-    pattern = re.compile(
-        # Match all numbers with respect to scientific notation
-        r"(?P<b>[+\-]?(?:0|[1-9]\d*)(?:\.\d*)?)(?:[eE](?P<e>[+\-]?\d+))?$")
-    result = pattern.search(string)
-    if result is not None:
-        number = float(result.group("b"))
-        exponent = result.group("e")
-        if exponent is not None:
-            number *= numpy.power(10, int(exponent)) if int(exponent) >= 0 \
-                else 1 / numpy.power(10, numpy.abs(int(exponent)))
-        return number
-    pi_pattern = re.compile(
-        # match pi with optional factor (e. g. "2.4pi")
-        r"(?P<f>[+\-]?(?:0|[1-9]\d*)(?:\.\d*)?)?(?:pi|π)$")
-    pi_result = pi_pattern.search(string)
-    if pi_result is not None:
-        factor = pi_result.group("f")
-        return numpy.pi if factor is None else numpy.pi * float(factor)
-    raise ValueError(_("No valid number specified"))
+    return _eval(ast.parse(string, mode="eval").body)
```

Comment:
A quick and dirty way to still allow the usage of pi, as well as to call powers with a ^. (The vast majority will never use the bitwise ^, but instead will use the ^ symbol for powers).


```suggestion
    string = string.replace("pi", f"({float(numpy.pi)})")
    string = string.replace("^", "**")
    return _eval(ast.parse(string, mode="eval").body)
```

Reply:
Ah, using brackets works well. I thought of a solution replacing it with `"*" + numpy.pi`, which resulted in errors when no factor was given, this is neat